src/components/ui/accordion.tsx

```diff
@@ -0,0 +1,92 @@
+'use client';
+
+import { forwardRef, ElementRef, ComponentPropsWithoutRef } from 'react';
+import {
+  Root,
+  Trigger,
+  Item,
+  Header,
+  Content,
+} from '@radix-ui/react-accordion';
+import { ChevronDown } from 'lucide-react';
+
+import { cn } from '@/lib/utils';
+
+const Accordion = Root;
+
+type AccordionTriggerProps = {
+  scheduleDate?: string;
+} & ComponentPropsWithoutRef<typeof Trigger>;
+
+const AccordionItem = forwardRef<
+  ElementRef<typeof Item>,
+  ComponentPropsWithoutRef<typeof Item>
+>(({ className, ...props }, ref) => (
+  <Item
+    ref={ref}
+    className={cn(
+      'border-b bg-primary-lightest rounded-xl p-4 shadow-md m-3',
+      className,
+    )}
+    {...props}
+  />
+));
+AccordionItem.displayName = 'AccordionItem';
+
+const AccordionTrigger = forwardRef<
+  ElementRef<typeof Trigger>,
+  AccordionTriggerProps
+>(({ className, children, scheduleDate, ...props }, ref) => (
+  <Header className="flex">
+    <Trigger
+      ref={ref}
+      className={cn(
+        'flex flex-1 items-center font-medium transition-all [&[data-state=open]>svg]:rotate-180',
+        className,
+      )}
+      {...props}
+    >
+      {children}
+      <div className="flex-1" />
+      {/* display only scheduleDate is set*/}
+      {scheduleDate && (
```

Comment:
The gap between the date and the icon is `6px`.
<img width="500" alt="image" src="https://github.com/Tascurator/tascurator-frontend/assets/144519746/3e9be360-d1d2-4de1-8453-72f380331cc1">